src/java/org/apache/ivy/plugins/signer/bouncycastle/OpenPGPSignatureGenerator.java

```diff
@@ -161,7 +159,7 @@ private PGPSecretKey readSecretKey(InputStream in) throws IOException, PGPExcept
                     key = k;
                 }
                 if ((keyId != null)
-                        && (Long.valueOf(keyId, 16).longValue() == (k.getKeyID() & MASK))) {
+                        && (Long.valueOf(keyId, 16) == (k.getKeyID() & MASK))) {
```

Comment:
I'm not 100% sure how Java autoboxing/unboxing deals in this case. Does it autobox the primitive to a wrapper `Long` and do a `==` check or does it do a unboxing of the `Long` to a primitive and do a `==` check?